### PR TITLE
Optimize code in RouteInfoManager

### PR DESCRIPTION
**Pull request by kang-hl**
### Which Issue(s) This PR Fixes

Fixes #issue_id

### CODE REVIEW

**Review by mxsm — COMMENTED** [review]
@kang-hl  Please check code style
image

**Comment by kang-hl** [comment]
> @kang-hl Please check code style image

Resolved.

**Review by joeCarf — COMMENTED** [review]
plz give more information and background of your change so i can help review :)
btw, there seems tests failures in ci

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ public void deleteTopic(final String topic, final String clusterName) {
             this.lock.writeLock().lockInterruptibly();
             //get all the brokerNames fot the specified cluster
             Set<String> brokerNames = this.clusterAddrTable.get(clusterName);
-            if (brokerNames == null || brokerNames.isEmpty()) {
+            if (brokerNames == null) {
```
Comment (joeCarf) [inline namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java:174]: why remove this?
suggest `CollectionUtils.isEmpty(brokerName)`

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java [hunk 2]
```diff
@@ -245,71 +245,65 @@ public RegisterBrokerResult registerBroker(
...
+                            //Remove the rejected brokerAddr from brokerLiveTable.
+                            brokerLiveTable.remove(new BrokerAddrInfo(clusterName, brokerAddr));
+                            return result;
+                        }
                     }
                 }
-            }
-
-            if (!brokerAddrsMap.containsKey(brokerId) && topicConfigWrapper.getTopicConfigTable().size() == 1) {
+            } else if (topicConfigWrapper.getTopicConfigTable().size() == 1) {
```
Comment (joeCarf) [inline namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java:291]: I think the former style is better

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java [hunk 3]
```diff
@@ -245,71 +245,65 @@ public RegisterBrokerResult registerBroker(
 
             boolean registerFirst = false;
 
-            BrokerData brokerData = this.brokerAddrTable.get(brokerName);
-            if (null == brokerData) {
+            BrokerData brokerData = ConcurrentHashMapUtils.computeIfAbsent((ConcurrentHashMap<String, BrokerData>) this.brokerAddrTable, brokerName, k -> new BrokerData(clusterName, brokerName, new HashMap<>()));
```
Comment (joeCarf) [inline namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java:248]: `this.brokerAddrTable.put(brokerName, brokerData);` this line is removed, the new BrokerData will not be put in brokerAddrTable

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java [hunk 4]
```diff
@@ -334,28 +328,42 @@ public RegisterBrokerResult registerBroker(
...
+                    } else {
+                        if (isPrimeSlave) {
+                            for (TopicConfig topicConfig : tcTable.values()) {
+                                if (this.isTopicConfigChanged(false, brokerName, topicConfig.getTopicName())) {
+                                    // Wipe write perm for prime slave
+                                    topicConfig.setPerm(topicConfig.getPerm() & (~PermName.PERM_WRITE));
+                                    this.createAndUpdateQueueData(brokerName, topicConfig);
+                                }
+                            }
+                        } else {
```
Comment (joeCarf) [inline namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java:361]: suggest not using too much if..else.., replace with just `if` clause

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java (hunk 1 above)
Comment (kang-hl) [reply namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java:174]: if (nameSet.isEmpty()) {
    this.clusterAddrTable.remove(clusterName);
    log.info("unregisterBroker, remove cluster from clusterAddrTable {}", clusterName);
}

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java (hunk 3 above)
Comment (kang-hl) [reply namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java:248]: computeIfAbsent

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java (hunk 4 above)
Comment (kang-hl) [reply namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java:361]: Moving `if` unrelated to the loop outside of the loop will result in better performance

**Pull request closed without merging**